src/PeoplePage.tsx

```diff
@@ -0,0 +1,74 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Loader } from './components/Loader';
+import { getPeople } from './api';
+import { Person } from './types';
+import { ErrorType } from './types/ErrorType';
+import { PeoplesTable } from './PeoplesTable';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<ErrorType | null>(null);
+
+  const loadPeople = useCallback(async () => {
+    setIsLoading(true);
+    try {
+      const response = await getPeople();
+
+      const peopleWithParentsInfo = response.map((person) => {
+        const mother = response.find(p => p.name === person.motherName);
+        const father = response.find(p => p.name === person.fatherName);
+
+        return {
+          ...person,
+          mother,
+          father,
+        };
+      });
+
+      setIsLoading(false);
+      if (response.length === 0) {
+        setError(ErrorType.noPeopleOnServer);
+      }
```

Comment:
I believe, disabling loading and zero-length array handling should be placed `finally` block